src/WebApi/Controllers/UsersController.cs

```diff
@@ -210,6 +210,37 @@ public Task<ActionResult<Result<CharacterCharacteristicsViewModel>>> GetCharacte
         }));
     }
 
+    /// <summary>
+    /// Get character competitive rating.
+    /// </summary>
+    /// <param name="id">Character id.</param>
+    /// <returns>The character rating.</returns>
+    /// <response code="200">Ok.</response>
+    [HttpGet("self/characters/{id}/competitive-rating")]
+    public Task<ActionResult<Result<CharacterCompetitiveRatingViewModel>>> GetCharacterRating([FromRoute] int id)
+    {
+        return ResultToActionAsync(Mediator.Send(new GetUserCharacterCompetitiveRatingQuery
+        {
+            UserId = CurrentUser.User!.Id,
+            CharacterId = id,
+        }));
+    }
+
+    /// <summary>
+    /// Get top character competitive ratings.
+    /// </summary>
+    /// <returns>The top character competitive ratings.</returns>
+    /// <response code="200">Ok.</response>
+    [HttpGet("leaderboard")]
+    public Task<ActionResult<Result<IList<CharacterCompetitiveRatingViewModel>>>> GetTopCharacterCompetitiveRatings()
+    {
+        return ResultToActionAsync(Mediator.Send(new GetTopUserCharactersByCompetitiveRatingQuery
+        {
+            // You can set the Top property if you want to customize the number of top characters returned.
+            // Top = 50, // for example
```

Comment:
Avoid adding code that is "useful for later" as we never know what's ahead of us. It's better to hardcode the value for now.